http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java

```diff
@@ -495,4 +496,16 @@ public HttpServerConfig setHttp2StreamIdleTimeout(Duration http2StreamIdleTimeou
         this.http2StreamIdleTimeout = http2StreamIdleTimeout;
         return this;
     }
+
+    public boolean isLegacyUriComplianceEnabled()
```

Comment:
This shouldn't be a config for end users, as the usage of this is specific to the application. If the application requires it, disabling the behavior via config would break the application. This should be configurable at the application level via `HttpServerBinder`, similar to virtual threads.